core/src/main/kotlin/io/fritz2/dom/listener.kt

```diff
@@ -0,0 +1,81 @@
+package io.fritz2.dom
+
+import io.fritz2.dom.html.Key
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.FlowPreview
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+import org.w3c.dom.Element
+import org.w3c.dom.HTMLInputElement
+import org.w3c.dom.HTMLSelectElement
+import org.w3c.dom.HTMLTextAreaElement
+import org.w3c.dom.events.Event
+import org.w3c.dom.events.KeyboardEvent
+import org.w3c.files.FileList
+
+/**
+ * [Action] contains the fired [Event] and targeting [Element]
+ */
+data class Action<E: Event, X: Element>(val event: E, val target: X)
```

Comment:
I think the name `Action` is wrong here. An action is, what you map your Event/Target/value to later on.
Also I do not think, that we need both seperated target and event. We can access target whenever needed from event and cast it. All we need is the type-information about the target.